extensions/amazon-lambda-alexa/deployment/src/main/java/io/quarkus/amazon/lambda/alexa/AlexaProcessor.java

```diff
@@ -0,0 +1,33 @@
+package io.quarkus.amazon.lambda.alexa;
+
+import java.util.Set;
+
+import org.reflections.Reflections;
+
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+
+import io.quarkus.deployment.annotations.BuildProducer;
+import io.quarkus.deployment.annotations.BuildStep;
+import io.quarkus.deployment.builditem.nativeimage.NativeImageProxyDefinitionBuildItem;
+import io.quarkus.deployment.builditem.nativeimage.ReflectiveClassBuildItem;
+import io.quarkus.deployment.pkg.steps.NativeBuild;
+
+public class AlexaProcessor {
+
+    @BuildStep(onlyIf = NativeBuild.class)
+    public void reflection(BuildProducer<ReflectiveClassBuildItem> reflectiveClass) {
+
+        Reflections reflections = new Reflections("com.amazon.ask");
+        Set<Class<?>> types = reflections.getTypesAnnotatedWith(JsonDeserialize.class);
+
+        types.forEach(type -> reflectiveClass
+                .produce(new ReflectiveClassBuildItem(true, true, type.getName(), type.getName() + "$Builder")));
```

Comment:
Can we try and use Jandex to do this? 
That is pretty much what all the other extensions do.

Reply:
Also, I think we should do that in the Jackson extension and just have this one depend on the Jackson extension.

Reply:
Yes, I was thinking it would be nice to have standard functionality that could automatically scan the class path and identify the JsonDeserialize usage.  The one caveat to this is that the Alexa ASK SDK also has a static Builder inner class, that requires ClassName$Builder to be added.

Reply:
Interestingly the Jackson extension potentially does have code to handle this & the builder already, yet it doesn't seem to pickup the JsonDeserialize annotation.

The combinedBuildIndex is small in size.  

```
  // handle the various @JsonDeserialize cases
        for (AnnotationInstance deserializeInstance : index.getAnnotations(JSON_DESERIALIZE)) {
            AnnotationTarget annotationTarget = deserializeInstance.target();
            if (CLASS.equals(annotationTarget.kind())) {
                DotName dotName = annotationTarget.asClass().name();
                if (!ignoredDotNames.contains(dotName)) {
                    addReflectiveHierarchyClass(dotName);
                }

                AnnotationValue annotationValue = deserializeInstance.value("builder");
                if (null != annotationValue && AnnotationValue.Kind.CLASS.equals(annotationValue.kind())) {
                    DotName builderClassName = annotationValue.asClass().name();
                    if (!BUILDER_VOID.equals(builderClassName)) {
                        addReflectiveHierarchyClass(builderClassName);
                    }
                }
            }
    } 
```

Reply:
You might need to create an `IndexDependencyBuildItem` in order to get that dependency into the inde (if it's not already in).

Reply:
Got this working with Jandex, using the IndexDependencyBuildItem as a BuildStep.

Code is quite simple now, all it does is draw in the Alexa dependencies, and then it uses the Jackson extension to do the work of identifying the JsonDeserialize & Builder classes.

Thanks to @stuartwdouglas for helping me today.

Reply:
Excellent!